tft_bot/helpers/screen_helpers.py

```diff
@@ -1,221 +1,145 @@
-        precision (float, optional): The precision to be used when matching the image. Defaults to 0.8.
+        path: The relative or absolute path to the image to be found.
+        precision: The precision to be used when matching the image. Defaults to 0.8.
+        offsets: A tuple of coordinates to off-set the region by. Useful if you only want to check a specific region.
+          All offsets are from the top-left of the game window.
 
     Returns:
-        (bool | list[int] | tuple[int, int]): The coordinates of the image if found within the specified region,
-            None otherwise.
+        The position of the image and it's width and height or None if it wasn't found
     """
-    try:
-        image = resource_path(image)
-        pos = imagesearch.imagesearcharea(image, x1, y1, x2, y2, precision)
-        count = 0
-
-        while pos[0] == -1:
-            time.sleep(timesample)
-            pos = imagesearch.imagesearcharea(image, x1, y1, x2, y2, precision)
-            count = count + 1
-            if count > max_samples:
-                break
-        return pos
-    except Exception:
-        return None
+    return get_on_screen(
+        window_title=CONSTANTS["window_titles"]["game"], path=path, precision=precision, offsets=offsets
+    )
 
 
-def find_image(path: str, precision: float = 0.8) -> None | list[int] | tuple[int, int]:
-    """Search for a given image, returning the coordinates if found.
+def get_on_screen(
+    window_title: str, path: str, precision: float = 0.8, offsets: tuple[int, int, int, int] | None = None
+) -> tuple[int, int, int, int] | None:
+    """
+    Check if a given image is detected on screen in a specific window's area.
 
     Args:
-        path (str): The relative or absolute path to the image to be found.
-        precision (float, optional): The precision to be used when matching the image. Defaults to 0.8.
+        window_title: The title of the window we should look at.
+        path: The relative or absolute path to the image to be found.
+        precision: The precision to be used when matching the image. Defaults to 0.8.
+        offsets: A tuple of coordinates to off-set the region by. Useful if you only want to check a specific region.
+          All offsets are from the top-left of the game window.
 
     Returns:
-        (None | list[int] | tuple[int, int]): The coordinates of the image if found within the specified region,
-            None otherwise.
+        The position of the image and it's width and height or None if it wasn't found
     """
-    try:
-        path = resource_path(path)
-        pos = imagesearch.imagesearch(path, precision)
-        return pos if pos[0] != -1 else None
-    except Exception:
+    window_bounding_box = get_window_bounding_box(window_title=window_title)
+    if not window_bounding_box:
         return None
 
+    image_to_find = cv2.imread(path, 0)
```

Comment:
I had been contemplating making the swap to cv2 directly since we were having the aforementioned issues with `python-imagesearch`, so thanks for getting the ball rolling on this!